libs/wire-api/src/Wire/API/Routes/Public/Galley.hs

```diff
@@ -333,8 +337,12 @@ type ConversationAPI =
            )
     :<|> Named
            "code-check"
-           ( Summary "Check validity of a conversation code"
+           ( Summary
+               "Check validity of a conversation code.\
+               \If the guest links team feature is disabled, this will fail with 404 CodeNotFound.\
+               \Note that this is currently inconsistent (for backwards compatibility reasons) with `POST /conversations/join` which responds with 409 GuestLinksDisabled if guest links are disabled."
                :> CanThrow CodeNotFound
+               :> CanThrow ConvNotFound
```

Comment:
@battermann as discussed off-line: this is thrown because the handler now does a conversation lookup where it didn't have to before, so this is all in order.  (which might mean that clients will have to make a trivial adjustment to their error handler.)